### PR TITLE
Combine poll and post

### PR DESCRIPTION
**Pull request by johandahlberg**
Combines starting a run on an arteria service and polling it til it finishes into a single action. This should make it easier for us to restart workflow from where they failed.

### CODE REVIEW

File: actions/poll_status.py [hunk 1]
```diff
@@ -25,17 +29,50 @@ def query(self, url, verify_ssl_cert):
...
+            self.logger.warning("An error was encountered when "
+                                "querying url: {0},  {1}".format(url, err))
+            return None
+
+    def post_to_endpoint(self, endpoint, body, irma_mode, verify_ssl_cert):
 
-    def run(self, url, sleep, ignore_result, verify_ssl_cert, max_retries=3):
+        def _rewrite_link(link):
+            endpoint_parsed = urlparse(endpoint)
+            first_part_of_path = endpoint_parsed.path.split('/')[1]
```
Comment (johanherman) [inline actions/poll_status.py]: We seem to define the destination URL upstream (in `group_vars/arteria-master/main.yml`) on the following format 

```irma_checksum_base_url: https://irma1.uppmax.uu.se:4444/arteria_checksum/api/1.0```

If we would change the way we define these URLs, by e.g. splitting out the host part and the path part, then we perhaps we might run the risk of having your `endpoint_parsed.path.split('/')[1]` returning an empty string if the full URL is concated together upstreams somewhere from e.g. `https://irma1.uppmax.uu.se:4444/` + `/arteria_checksum/api/1.0` into `https://irma1.uppmax.uu.se:4444//arteria_checksum/api/1.0`. 

Although I guess it is not very likely that we would end up writing the URLs in that way (compare to file paths though, where it is much more likely that you have a trailing `/` to indicate that it is a folder). And perhaps `urlparse()` also fixes this for you. So it is probably just a moot point anyway.

File: actions/poll_status.py [hunk 2]
```diff
@@ -25,17 +29,50 @@ def query(self, url, verify_ssl_cert):
...
+            link_parsed = urlparse(link)
+            return "{}://{}/{}{}?{}".format(endpoint_parsed.scheme,
+                                            endpoint_parsed.netloc,
+                                            first_part_of_path,
+                                            link_parsed.path,
+                                            endpoint_parsed.query)
```
Comment (johanherman) [inline actions/poll_status.py]: At the moment we are just passing the `apikey` as an argument on the URL, but I guess that could change someday, somehow. Hopefully the destination microservice should behave nicely even with unexpected queries arrive, but you never know.

**Comment by johandahlberg** [comment]
I have addressed your comments, with exception for the query part, which I think we will have to look into if that becomes a problem in the future. Let me know if you thinks it's alright to merge, or if you have any additional comments.

**Comment by johanherman** [comment]
:+1:

**Pull request merged**